airbyte-api/src/main/openapi/config.yaml

```diff
@@ -1723,6 +1723,31 @@ paths:
           $ref: "#/components/responses/NotFoundResponse"
         "422":
           $ref: "#/components/responses/InvalidInputResponse"
+  /v1/source_oauths/complete_oauth_return_secret:
+    post:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CompleteSourceOauthRequest"
+        required: true
+      tags:
+        - source_oauth
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SecretId"
+          description: Successful operation
+        "404":
+          $ref: "#/components/responses/NotFoundResponse"
+        "422":
+          $ref: "#/components/responses/InvalidInputResponse"
+      operationId: completeSourceOAuthReturnSecret
+      summary: >-
+        Given a source def ID generate an access/refresh token and save as a secret. Return the secret
+        ID.
```

Comment:
Some destinations have oAuth as well

Reply:
Yup, we recognize that there is destination oauth as well but that doesn't serve the primary use case we need this endpoint for (Powered by Airbyte headless flow) so this is the one we what's to get in for q1. I think at some point we do plan to have a similar endpoint but for destinations.